### ISSUE TITLE
Bad error message for too-long reads

### ISSUE THREAD

**Issue by YasirKusay** [amplab/snap#138]
Hi, I am using snap-aligner to run this command:
`snap-aligner single index input.fq -o output.sam `

and have got this error:

```
Loading index from directory... 0s.  86283964 bases, seed size 20
Aligning.
Failed to write into fresh buffer; trying providing the -wbs switch with a larger value
SNAP exited with exit code 1 from line 236 of file SNAPLib/ReadWriter.cpp
```

Some context: index is a small index created from viral.1.1.genomic.fna and input.fq is an assembly file (also relatively small). In addition, the assembly file was originally outputted in fasta format but I have converted it to fastq using dummy quality values (composed of '#' characters). 

Can you help me out please?

**Comment by bolosky** [comment]
That happens when the set of reads written by a single input read pair is too big to fit into an output buffer (which is 16MB by default).  The normal reason you'd see this is if a read pair produced lots of alternate alignments, but I suppose it could also happen if you made really big reads or something.

You could try rerunning snap giving it the -wbs switch with some value larger than the default 16 and see if that helps.

What version are you running?  It doesn't look like the current one based on the line number where it exited.  I think you might have a pretty old one, so you might want to update to the newest 1.0.4 version.

Also, if it still doesn't work, send me the command line you used to run snap.

--Bill


From: YasirKusay ***@***.***>
Sent: Saturday, November 6, 2021 4:46 AM
To: amplab/snap ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [amplab/snap] -wbs Issue (Issue #138)


Hi, I am using snap-aligner to run this command:
snap-aligner single index input.fq -o output.sam

and have got this error:

Loading index from directory... 0s.  86283964 bases, seed size 20

Aligning.

Failed to write into fresh buffer; trying providing the -wbs switch with a larger value

SNAP exited with exit code 1 from line 236 of file SNAPLib/ReadWriter.cpp

Some context: index is a small index created from viral.1.1.genomic.fna and input.fq is an assembly file (also relatively small). In addition, the assembly file was originally outputted in fasta format but I have converted it to fastq using dummy quality values (composed of '#' characters).

Can you help me out please?

-
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Famplab%2Fsnap%2Fissues%2F138&data=04%7C01%7Cbolosky%40microsoft.com%7C5f1b1726bf4b444b65a408d9a11b0dca%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637717959849080885%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=x6Z1gl0H6po%2B7eUl4%2FUjUtKib9D%2FZFhaou6MZ12M1fQ%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAHPTWKLLQ7ZCCXY6HPXYJLUKUIQ3ANCNFSM5HPQZBOQ&data=04%7C01%7Cbolosky%40microsoft.com%7C5f1b1726bf4b444b65a408d9a11b0dca%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637717959849090833%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2FJg9T0%2FXTYMfZX98945Dp%2BI5WBa3h4xmRJ%2F9De0izFg%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cbolosky%40microsoft.com%7C5f1b1726bf4b444b65a408d9a11b0dca%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637717959849090833%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=jXry6U2GVIxKkWoVhDV9U09wQ3KlszDauqi1R2k%2B9rQ%3D&reserved=0> or Android<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cbolosky%40microsoft.com%7C5f1b1726bf4b444b65a408d9a11b0dca%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637717959849100791%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=tGfhlskFebtDmpPgFr8UG7Pnr2QzQalUEFO0iTglRl8%3D&reserved=0>.

**Comment by YasirKusay** [comment]
Hi @bolosky, thank you for your reply. The read size are no greater than 700 bp for the sample I am using and I am using version 1.0.23 beta. While I did try to increase the size of -wbs (to 184), to no success, I will try again but with larger computational resources.

**Comment by bolosky** [comment]
No, something else is going on.  You shouldn't need more computational resources for that.

I'd try running the current version of SNAP (1.0.4).  If that doesn't work, then if you can share your reference and read sets for me I'll try to repro it and see what the problem.

**Comment by YasirKusay** [comment]
I will try that, thank you

**Comment by YasirKusay** [comment]
Hi @bolosky , I installed SNAP version 1.0.4 (although I am greeted with "Welcome to SNAP version 1.0.3" when just running `snap-aligner`). 
Unfortunately, I am still getting the same issue despite increasing the -wbs size. Can you help me out?

**Comment by bolosky** [comment]
Sure.  Can you give me access to your input files?  I'll try to repro it here and see what's happening.

If you don't want to broadcast the files, you can email links directly to me at ***@***.***

--Bill

**Comment by YasirKusay** [comment]
Hi, thank you for your reply. I would like to send you an email but your email address is hidden.

**Comment by bolosky** [comment]
My last name @ microsoft.com.

**Comment by bolosky** [comment]
What's happening here is that the input has at least one read longer than the compiled-in maximum read length and it's not properly producing an error message.  We need to fix the error handling.

**Comment by YasirKusay** [comment]
Hi Bill,

Thank you for your reply, I will now attempt to run it again with a larger read value.

Kind Regards,

Yasir

________________________________

**Comment by bolosky** [comment]
Fixed in 2.0
